### PR TITLE
Various fixes and changes

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
Done a few minor additions and changes over the last week. So this will rather be a miscellaneous PR for now.
I am marking this as a draft since there may be more to come before this should be merged.
But it's here in case you wanna have a look at it already (I honestly don't fully remember what i changed either, so this is a great oppurtunity for me to review my changes again).

Anyway, here's what I changed:
* I fixed some of the formatting issues from my earlier PRs, switched back to tabs and I think I matched your brackets style and similar too, this will definitely be more coherent at least. (Warning there will be lots of whitespace changes because of switching from spaces to tabs, luckily GitHub has an option to omit those from the changes screen)
* Added PotionMetaMock (very very similar to the SuspiciousStewMetaMock)
* Improved Player deaths, exp and health will now be reset too
* Added PlayerMock#respawn() which simulates a respawn, it will reset the location, call an event and more
* Fixed ServerMock#getBukkitVersion() (since that seemed to still be stuck at 1.12), also made it a static constant so its easier to see when that needs to be updated
* Implemented the "cursor item" for InventoryViews/Players.

### CODE REVIEW

**Comment by seeseemelk** [comment]
Oops, I accidentally marked it as no longer a draft when I meant to add a review

**Comment by TheBusyBiscuit** [comment]
Hmmm, I am sure there is a way to convert it back to a draft somewhere... Can't find it though. Oh well, it's not like that mattered much anyway. Guess it is a proper PR now XD

File: src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java [hunk 1]
```diff
@@ -162,23 +155,22 @@ public ItemStack addItem(ItemStack item)
 				items[i] = item.clone();
 				items[i].setAmount(toAdd);
 				item.setAmount(item.getAmount() - toAdd);
-			}
-			else if (item.isSimilar(oItem) && oItem.getAmount() < oItem.getMaxStackSize())
+			} else if (item.isSimilar(oItem) && oItem.getAmount() < oItem.getMaxStackSize())
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java]: This `else` should be on a separate line

File: src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java [hunk 2]
```diff
@@ -209,183 +201,184 @@ public void setContents(ItemStack[] items)
 			if (i < items.length && items[i] != null)
 			{
 				this.items[i] = items[i].clone();
-			}
-			else
+			} else
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java]: This `else` should be on a separate line

File: src/main/java/be/seeseemelk/mockbukkit/inventory/ItemFactoryMock.java [hunk 3]
```diff
@@ -66,8 +68,7 @@ public ItemMeta getItemMeta(Material material)
 		try
 		{
 			return getItemMetaClass(material).newInstance();
-		}
-		catch (InstantiationException | IllegalAccessException e)
+		} catch (InstantiationException | IllegalAccessException e)
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/ItemFactoryMock.java]: `catch` should be on a separate line

File: src/main/java/be/seeseemelk/mockbukkit/inventory/ItemFactoryMock.java [hunk 4]
```diff
@@ -92,8 +93,7 @@ public boolean equals(ItemMeta meta1, ItemMeta meta2)
 		if (meta1 != null && meta2 != null)
 		{
 			return meta1.equals(meta2);
-		}
-		else
+		} else
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/ItemFactoryMock.java]: `else` should be on a separate line

File: src/main/java/be/seeseemelk/mockbukkit/inventory/ItemFactoryMock.java [hunk 5]
```diff
@@ -111,16 +111,20 @@ public ItemMeta asMetaFor(ItemMeta meta, Material material)
...
+				}
 			}
-			
-			throw new NoSuchMethodException("Cannot find an ItemMeta constructor for the class \"" + meta.getClass().getName() + "\"");
-		}
-		catch (SecurityException | InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e)
+
+			throw new NoSuchMethodException(
+					"Cannot find an ItemMeta constructor for the class \"" + meta.getClass().getName() + "\"");
+		} catch (SecurityException | InstantiationException | IllegalAccessException | InvocationTargetException
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/inventory/ItemFactoryMock.java]: Catch should be on a separate line

**Review by seeseemelk — APPROVED** [review]
Given that you said that you tried to keep the style consistent, I've made a few comments on places where you missed the reformatting. I've ignored this in all earlier pull requests as I felt it was much more important to get more features into the project rather than some nice consistent coding style.

**Comment by TheBusyBiscuit** [comment]
Thanks! I adjusted my formatter and fixed the issues you pointed out.
Meanwhile I have also found that damned "Convert back to draft" button XD

**Comment by TheBusyBiscuit** [comment]
Alright, done the changes to Player/OfflinePlayer mocks as discussed earlier.
ServerMock#getOfflinePlayer() will now return a PlayerMock if the Player is online, the cached version if the Player was ever on the server and will instantiate a new OfflinePlayerMock otherwise.
Also added an OfflinePlayerMock#join() method.
There's still a few things I wanna do before I mark this as "Ready for review" but this covers most of the things on my todo list.

**Comment by TheBusyBiscuit** [comment]
Alright, will probably call it a day here and mark it as "Ready for review". Pretty much done all my todos for this one and I don't really wanna stretch this already very large PR too much.

**Comment by TheBusyBiscuit** [comment]
BTW to elaborate the changes to the Permissions:

Your current implementation resembled "isPermissionSet" whereas Bukkit actually tries to get the Permission object for that name rather than just checking if they have this specific name.

This also had no impact on the test outcome but makes it more in-line with the original implementations.

**Comment by seeseemelk** [comment]
I never used permissions all that much, so it's good that you caught that! Thanks!

**Pull request merged**